### PR TITLE
fix(column): 'statuscolumn' not drawn after virt_lines with "n" in 'cpo'

### PR DESCRIPTION
**Pull request by luukvbaal**
Problem:    The 'statuscolumn' is not drawn and the line itself is drawn
            at an offset to the rest of the buffer after virt_lines if
            'cpoptions' includes "n".
Solution:   Make sure 'statuscolumn' is drawn.

### CODE REVIEW

File: src/nvim/drawline.c [hunk 1]
```diff
@@ -2967,16 +2967,17 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool nochange,
...
       if (statuscol.draw) {
-        if (wlv.row == startrow + wlv.filler_lines + 1
-            || wlv.row == startrow + wlv.filler_lines) {
-          // Re-evaluate 'statuscolumn' for the first wrapped row and non filler line
-          statuscol.textp = NULL;
-        } else if (statuscol.textp) {
+        if (wlv.row == startrow + wlv.filler_lines) {
+          statuscol.textp = NULL;  // re-evaluate for first non-filler line
+        } else if (vim_strchr(p_cpo, CPO_NUMCOL) && wlv.row > startrow + wlv.filler_lines) {
+          statuscol.draw = false;  // don't draw status column if "n" is in 'cpo'
```
Comment (zeertzjq) [inline src/nvim/drawline.c:2973]: Hmm, sign and fold columns are still needed on wrapped screen lines with `cpo-n`.

Comment (luukvbaal) [reply src/nvim/drawline.c:2973]: This is for `cpo+n` (the old comment was incorrect).

Comment (zeertzjq) [reply src/nvim/drawline.c:2973]: Yes. They are still needed.

Comment (luukvbaal) [reply src/nvim/drawline.c:2973]: Yes, they are drawn by falling back the default columns by setting `statuscol.draw` to false.

Comment (luukvbaal) [reply src/nvim/drawline.c:2973]: Or should be, did you observe that this is not the case?

Comment (zeertzjq) [reply src/nvim/drawline.c:2973]: > Yes, they are drawn by falling back the default columns by setting `statuscol.draw` to false.

Hmm, won't that look a bit inconsistent?

Comment (luukvbaal) [reply src/nvim/drawline.c:2973]: It might, this was the compromise I decided on in the original PR.

Comment (luukvbaal) [reply src/nvim/drawline.c:2973]: The alternative is placing the burden of making it look right on the user when `v:virtnum > 0`.

Comment (luukvbaal) [reply src/nvim/drawline.c:2973]: But that would not even be possible, unless if we don't pad the statuscolumn with whitespace for wrapped lines and `cpo+=n`.

Currently a shorter returned string would simply be padded to the maximum statuscolumn width, nullifying `cpo+=n`.

**Pull request merged**